### ISSUE TITLE
Cody context selector UI is confusing in an App context and impedes onboarding

### ISSUE THREAD

**Issue by toolmantim** [sourcegraph/sourcegraph#53844]
In the Cody App onboarding the Chat UI is the final all-important step where they try Cody for real, and the context selector UI presents some big hurdles for people to get started.

```[tasklist]
### Suggested fixes
- [ ] Include all repos in the context selector inferred list (p0)
- [ ] Remove context selector search (p0)
- [ ] Fix embeddings tooltip (p0)
- [ ] Add indexing status (p1)
```

### Issues

1. It doesn't include any repositories by default

https://github.com/sourcegraph/sourcegraph/assets/153/8ffd5dbe-8db7-4a2b-9afb-69a268eb8eba

2. The search filtering isn't working as expected

https://github.com/sourcegraph/sourcegraph/assets/153/7db12346-75e9-4ca5-8ba8-7f6ab94cde5a

3. The context selector repo UI copy and design misleads people into thinking it's where you manage your repositories

<img width="1136" alt="Screenshot 2023-06-21 at 6 41 58 pm" src="https://github.com/sourcegraph/sourcegraph/assets/153/3d11d751-69db-49f1-9aae-f1c13c4ca1b3">

4. When you finally enable a repo in the chat, the embedding status tells people to enable embeddings (but it will be automatically kicked off… people just need to wait for it to finish):

<img width="1136" alt="Screenshot 2023-06-21 at 6 47 13 pm" src="https://github.com/sourcegraph/sourcegraph/assets/153/82b1a510-22b1-46b5-a611-c824aba1a64a">

### Include all repos in the inferred list of the context selector, and remove search (p0)

<img width="350" alt="Screenshot 2023-06-21 at 6 59 57 pm" src="https://github.com/sourcegraph/sourcegraph/assets/153/9170d386-c007-421f-86a6-c6b2aed8f00c">

<img width="357" alt="Screenshot 2023-06-21 at 7 01 13 pm" src="https://github.com/sourcegraph/sourcegraph/assets/153/5b011d58-006b-4c54-be83-3cd79eb4f875">

This ensures they're on by default, so a user can easily start chatting about any of their repositories, as well as disabling any repos if they want to focus on one. There's no more search, or confusing mentions of "Adding repositories…".

_(something to investigate later: look into making it easy to switch between single repos)_

### Improve the indexing status tooltip

https://github.com/sourcegraph/sourcegraph/blob/2a7c0b9d8a5480ce609dc8e9ee6801207fc13098/client/web/src/cody/components/ScopeSelector/RepositoriesSelectorPopover.tsx#L484

with:

```js
<>Please wait while this repository is being indexed. See the indexing status in <Link to="...">repository settings</Link>.</>
```

(see https://github.com/sourcegraph/sourcegraph/blob/2a7c0b9d8a5480ce609dc8e9ee6801207fc13098/client/web/src/repo/tree/TreePageContent.tsx#L232 for another example of a tooltip with link)

### Add indexing status UI

We should look to include some kind of progress indicator UI in the main chat UI, outside of the context selector. It could reuse the code that comes out of https://github.com/sourcegraph/sourcegraph/issues/53809.

For a design reference, there’s [VS Code’s design for this (but it's single-repo)](https://www.figma.com/file/9112BsKsJc1BpO2j8XYwLL/%F0%9F%A4%96-Cody-VS-Code-%5BMain%5D?type=design&node-id=3280-144903&t=q44U066JzHWxkPun-4) and I've also just done [a quick concept](https://www.figma.com/file/ZkJFUvVzZW9TgFg7k9w9Oq/App-%5BMAIN%5D?type=design&node-id=405-1752&t=B1BEbKqFvzOVYMYi-4) off @danielmarquespt’s design for multi-repo:

<img width="933" alt="Screenshot 2023-06-21 at 7 50 08 pm" src="https://github.com/sourcegraph/sourcegraph/assets/153/94ccb473-8520-411f-9343-91a34820aa42">

## References

See Slack conversation: https://sourcegraph.slack.com/archives/C01U8SJUKJM/p1687332618711079?thread_ts=1687226876.693649&cid=C01U8SJUKJM

**Comment by toolmantim** [comment]
@danielmarquespt did you ever end up finalising a design for a multi-repo embeddings status UI footer for app’s onboarding? I added a quick concept above, but just wondering if that's still planned to land in the release?

**Comment by toolmantim** [comment]
@rrhyne time is super critical here… so I'm not sure if I agree we should change course from my suggestion.

> I don't think selecting all repositories on by default is the right answer, as it could reduce the accuracy of the response.

Are you suggesting that we start with no repositories selected? You still have a discoverability problem, where you have to remember the name of the repos that are setup (even though there are only 10) because the UI is a blank search.

> Removing is fine for App, but hurts enterprise customers. @thenamankumar can we try to fix the search issue?

@rrhyne I'm confused… these things would be conditional based on App, so shouldn't hurt Enterprise customers? The search issue doesn't appear to exist on Enterprise.

> The text on the selector states only: "Add repositories" when the selector is empty. "Add repositories to context" would be more accurate.

If we keep it empty (which I think would be a mistake for onboarding), I would suggest a simple "Choose a repository…" label.

@rrhyne are you sure you want to change course from my suggested solution above?

**Comment by rrhyne** [comment]
Personally, I think the solution is to show all 10 as you suggest in app only… enterprise should show repos you contribute to (but that won’t happen for this release). But not select them. Then we use Paulo’s ui popover to help users know they need to select a repo.

**Comment by rrhyne** [comment]
Update: I synced with Tim and with time constraints and the likelyhood that not all users will have 10 repos, his fix is the right call for app. 

We also have the benefit of iterating quickly with app, so we should follow with a better solution and continue our user testing to verify its working.

**Comment by toolmantim** [comment]
👍🏼 I don't know if we have data yet, but I'm guessing most users will add 1 or a few repos to start, so at least for the most basic use cases they'll have something selected for their first chat experience.

**Comment by toolmantim** [comment]
Another quick fix we could add is not let a user type a message without selecting a repository. And even letting go of multi-repo for app to start, and it being just the equivalent of a repository select field.

E.g. similar to if the repo selector was just a `<select required="true">`, and they get a browser form validation error when trying the submit a message to Cody.

**Comment by almeidapaulooliveira** [comment]
Is this the experience we want it? I don't think selecting a repository is a hard requirement to use the chat. 

I think the selector it's an optional part of the flow, and users must understand what it does, and why they should use it. 

**Here's a parallel with other chat interfaces:**
Imagine if WhatsApp (or other chat) required users to set the context of a conversation before starting a chat with a friend. Why? Sometimes you begin a conversation without knowing where it will lead.

During the conversation, you may feel the need to share more. A picture, a document, a GPS location, etc. (_That's parallel with giving more context_). That makes sense. Users have a goal, a job to do. From then on, they will look over the interface in how to do it.

**Comment by toolmantim** [comment]
> Is this the experience we want it? I don't think selecting a repository is a hard requirement to use the chat.

Agreed, we should ignore my suggestion of a required form field and stick to the plan. But let’s keep the repositories default enabled until the blank chat experience is smarter (like a human would be in WhatsApp).

**Comment by toolmantim** [comment]
I've removed the suggested fixes from the body of the issue no we've got issues created, and also put together designs for both VS Code and App to make sure we know where we're heading, and to make it consistent between App and VS Code:

[Figma of the simplified Chat Context selector, including all tooltip and repo icon states](https://www.figma.com/file/9112BsKsJc1BpO2j8XYwLL/%F0%9F%A4%96-Cody-VS-Code-%5BMain%5D?type=design&node-id=3863-147807&mode=design&t=lX2Dgr2mFQjMJ3fI-4):

<img width="3677" alt="Repository indexing states" src="https://github.com/sourcegraph/sourcegraph/assets/153/adef9d8f-057d-4357-aab0-a2cb8332f222">

[Figma of the indexing progress bars](https://www.figma.com/file/9112BsKsJc1BpO2j8XYwLL/%F0%9F%A4%96-Cody-VS-Code-%5BMain%5D?type=design&node-id=3880-144460&mode=design&t=lX2Dgr2mFQjMJ3fI-4):

<img width="5201" alt="Repository indexing progress banners" src="https://github.com/sourcegraph/sourcegraph/assets/153/fb2ee256-12cf-4288-bc4f-327cdbaff584">

The parts that are important for 5.1 launch are the suggested fixes in the body still… and it's going to be whatever we can land for launch based on what you can do @thenamankumar 🙏🏼 But these designs should be used to fill in any details such as tooltip copy, icon states, etc.

**Comment by toolmantim** [comment]
(I have to run for the day, so didn't make an effort to organise those issues and update any based on the above designs)
